libs/parallelism/algorithms/include/hpx/parallel/container_algorithms/adjacent_find.hpp

```diff
@@ -348,12 +347,16 @@ namespace hpx { namespace ranges {
         {
             using iterator_type =
                 typename hpx::traits::range_traits<Rng>::iterator_type;
-            using is_segmented =
-                hpx::traits::is_segmented_iterator<iterator_type>;
-            return hpx::parallel::v1::detail::adjacent_find_(
-                hpx::execution::seq, std::begin(rng), std::end(rng),
-                std::forward<Pred>(pred), std::forward<Proj>(proj),
-                is_segmented());
+
+            static_assert(
+                (hpx::traits::is_forward_iterator<iterator_type>::value),
```

Comment:
Should I change this to input iterator?

Reply:
Nope if you check the [C++ working draft](http://eel.is/c++draft/alg.adjacent.find), it is a forward iterator for adjacent_find

Reply:
Ohh ok thanks! I'll be sure to refer the C++ working draft from next time.